pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -420,7 +434,10 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 		rcertificatemanagement.CertificateManagement(&rcertificatemanagement.Config{
 			Namespace:       render.IntrusionDetectionNamespace,
 			ServiceAccounts: []string{render.IntrusionDetectionName},
```

Comment:
You also need to add the service account `ADAPIObjectName` to it. This is the service account that will create a CertificateRequest when CertificateManagement is enabled. If the service account is missing here, it will not get the right RBAC for that operation.

Reply:
https://github.com/tigera/operator/pull/1931/commits/6ebf6fd366dd2d8e0d814b828b68b4a8af56d9a4